go/test/endtoend/backup/vtbackup/backup_only_test.go

```diff
@@ -0,0 +1,304 @@
+				path.Join(os.Getenv("VTDATAROOT"), "tmp", "backupstorage"),
+				"RemoveBackup", shardKsName, scanner.Text(),
+			)
+			assert.Nil(t, err)
+		}
+	}
+
+}
+
+func initTablets(t *testing.T, startTablet bool, initShardMaster bool) {
+	// Initialize tablets
+	for _, tablet := range []cluster.Vttablet{*master, *replica1} {
+		err := localCluster.VtctlclientProcess.InitTablet(&tablet, cell, keyspaceName, hostname, shardName)
+		assert.Nil(t, err)
+
+		if startTablet {
+			err = tablet.VttabletProcess.Setup()
+			assert.Nil(t, err)
+		}
+	}
+
+	if initShardMaster {
+		// choose master and start replication
+		err := localCluster.VtctlclientProcess.InitShardMaster(keyspaceName, shardName, cell, master.TabletUID)
+		assert.Nil(t, err)
+	}
+}
+
+func restore(t *testing.T, tablet *cluster.Vttablet, tabletType string, waitForState string) {
+	// Erase mysql/tablet dir, then start tablet with restore enabled.
+
+	log.Info("restoring tablet %s", time.Now())
+	resetTabletDirectory(t, *tablet, true)
+
+	err := tablet.VttabletProcess.CreateDB(keyspaceName)
+	assert.Nil(t, err)
+
+	// Start tablets
+	tablet.VttabletProcess.ExtraArgs = []string{"-db-credentials-file", dbCredentialFile}
+	tablet.VttabletProcess.TabletType = tabletType
+	tablet.VttabletProcess.ServingStatus = waitForState
+	tablet.VttabletProcess.SupportsBackup = true
+	err = tablet.VttabletProcess.Setup()
+	assert.Nil(t, err)
+}
+
+func resetTabletDirectory(t *testing.T, tablet cluster.Vttablet, initMysql bool) {
+
+	extraArgs := []string{"-db-credentials-file", dbCredentialFile}
+	tablet.MysqlctlProcess.ExtraArgs = extraArgs
+
+	// Shutdown Mysql
+	err := tablet.MysqlctlProcess.Stop()
+	assert.Nil(t, err)
+	// Teardown Tablet
+	err = tablet.VttabletProcess.TearDown()
+	assert.Nil(t, err)
+
+	// Empty the dir
+	err = os.RemoveAll(tablet.VttabletProcess.Directory)
+	assert.Nil(t, err)
+
+	if initMysql {
+		// Init the Mysql
+		tablet.MysqlctlProcess.InitDBFile = newInitDBFile
+		err = tablet.MysqlctlProcess.Start()
+		assert.Nil(t, err)
+	}
+
+}
+
+func tearDown(t *testing.T, initMysql bool) {
+	// reset replication
+	promoteSlaveCommands := "STOP SLAVE; RESET SLAVE ALL; RESET MASTER;"
+	disableSemiSyncCommands := "SET GLOBAL rpl_semi_sync_master_enabled = false; SET GLOBAL rpl_semi_sync_slave_enabled = false"
+	for _, tablet := range []cluster.Vttablet{*master, *replica1, *replica2} {
+		_, err := tablet.VttabletProcess.QueryTablet(promoteSlaveCommands, keyspaceName, true)
+		assert.Nil(t, err)
+		_, err = tablet.VttabletProcess.QueryTablet(disableSemiSyncCommands, keyspaceName, true)
+		assert.Nil(t, err)
+		for _, db := range []string{"_vt", "vt_insert_test"} {
+			_, err = tablet.VttabletProcess.QueryTablet(fmt.Sprintf("drop database if exists %s", db), keyspaceName, true)
+			assert.Nil(t, err)
+		}
+	}
+
+	for _, tablet := range []cluster.Vttablet{*master, *replica1, *replica2} {
+		//Tear down Tablet
+		//err := tablet.VttabletProcess.TearDown()
+		//assert.Nil(t, err)
```

Comment:
If this is no longer relevant, delete instead of commenting out.

Reply:
I am not 100% sure. 
As we are restarting the mysql here the rest of the code is not at all required. Like `promoteSlaveCommands`  & `disableSemiSyncCommands `

Ideally, we will have to uncomment the commented code and comment `resetTabletDirectory(t, tablet, initMysql)`. But that doesn't work

Reply:
Let's get this merged and investigate that separately.